cli/serve.go

```diff
@@ -38,24 +39,42 @@ func handler(_ *cobra.Command, _ []string) error {
 
 	// https://golang.org/pkg/os/signal/#Notify
 	// should be of buffer size at least 1
-	c := make(chan os.Signal, 1)
-	signal.Notify(c, os.Interrupt, syscall.SIGTERM, syscall.SIGINT)
+	shutdownHandler := make(chan os.Signal, 2)
+	signal.Notify(shutdownHandler, os.Interrupt, syscall.SIGTERM, syscall.SIGINT)
+
+	stop := make(chan struct{}, 2)
+
+	go func() {
+		// first catch - stop the container
+		<-shutdownHandler
+		// send signal to stop execution
+		stop <- struct{}{}
+
+		// after first hit we are waiting for the second
+		// second catch - exit from the process
+		<-shutdownHandler
```

Comment:
Can we write a message "exit() forced" or something like that?

Reply:
Also, can we write something like - received stop request, waiting to complete for %s(unlimited) seconds (based on graceful config setup). It will let people know that RR didn't stuck but rather doing something.

Reply:
> See attached comments. In general:
> 
>     1. Defaults are still too aggressive.
> 
>     2. Default endure logging is too verbose.
> 
>     3. Lost the ability to make your own main.go file with your own plugins. (OR, I assume we can simply call Container. Register again?)
> 
>     4. It would be ideal to add few help messages for users since graceful shutdown is not a super obvious feature.



> See attached comments. In general:
> 
>     1. Defaults are still too aggressive.
> 
>     2. Default endure logging is too verbose.
> 
>     3. Lost the ability to make your own main.go file with your own plugins. (OR, I assume we can simply call Container. Register again?)
> 
>     4. It would be ideal to add few help messages for users since graceful shutdown is not a super obvious feature.

1. The grace period of the 30s is the common practice.
2. Agree about log level, fixed. 
3. Own plugins can be placed to the `root.go` without sharing the `Container` instance between `main.go` <-> `root.go` which is totally confusing. Now, all parts of the system properly initializing in one place.
4. Agree, fixed -> `2021/03/07 13:31:00 stop signal received, grace timeout is: 30 seconds` + `exit forced` on the second signal.